libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt

```diff
@@ -8,6 +8,15 @@ import com.mapbox.navigation.base.typedef.TimeFormatType
 
 const val DEFAULT_NAVIGATOR_POLLING_DELAY = 1500L
 
+/**
+ * Defines navigation options
+ *
+ * @param roundingIncrement defines threshold in meters for determination is user off the route
+ * @param timeFormatType defines time format for calculation remaining trip time
+ * @param navigatorPollingDelay defines how often we will request navigation status updates
```

Comment:
Let's refactor to something like: "defines the interval of navigation status updates which provide RouteProgress and Enhanced location updates". Possibly with a link to where to register those listeners.

Reply:
`navigatorPollingDelay` https://github.com/mapbox/mapbox-navigation-android/blob/5315386bca169ecb144b9862de33a22bdb6a23a5/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L267 is different from `STATUS_POLLING_INTERVAL` https://github.com/mapbox/mapbox-navigation-android/blob/5315386bca169ecb144b9862de33a22bdb6a23a5/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L39 which definitely needs to be configurable and not hardcoded (but that's a different story and we can fix in a future PR). For `navigatorPollingDelay` we can use our previous documentation 👀 https://github.com/mapbox/mapbox-navigation-android/blob/5315386bca169ecb144b9862de33a22bdb6a23a5/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationConstants.kt#L112-L122